codex/Main.hs

```diff
@@ -75,7 +76,11 @@ writeCacheHash cx = writeFile $ hashFile cx
 
 update :: Bool -> Codex -> Builder -> IO ()
 update force cx bldr = displayConsoleRegions $ do
-  (mpid, dependencies, workspaceProjects') <- resolveCurrentProjectDependencies bldr $ hackagePath cx </> "00-index.tar"
+  (mpid, dependencies, workspaceProjects') <- case bldr of
```

Comment:
Since I have no idea about stacks hacks, I can only say this is entirely correct for Cabal, so I left it unchanged. You may want to take the filename of [hackageTarball](https://hackage.haskell.org/package/hackage-db-2.0.1/docs/Distribution-Hackage-DB-Path.html#v:hackageTarball) and use that for stack as well.